### PR TITLE
Restrict hero block and remove collegiate split

### LINKED ISSUE #2517: Remove collegiate split
# Story

As a developer I want to remove the collegiate split to reduce complexity in managing the codebase and considerations for testing.

# Proposed solution

- [x] Investigate how removing Collegiate split from a site like uiowa.edu affects the ability to manage existing pages.
- [x] If necessary based on previous investigation, audit usage of blocks that are available in collegiate split and not in default profile.
- [x] Handle adding limits to main menu block not based on collegiate split: https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/layout_builder_custom/layout_builder_custom.module#L200
- [x] Turn off collegiate split in existing sites
- [x] Remove the `collegiate` directory from `/config/features`

### LINKED ISSUE #2585: Restrict hero block creation
# Story

As a project maintainer I want to limit user's ability to create new hero blocks so that people get used to using the banner instead.

# Proposed solution

- [ ] Restrict usage of hero block in LB restrictions for page display.
- [ ] Copy those settings to other page `entity_view_display` instances.

### PR DESCRIPTION
**Pull request by pyrello**
# Updates
Resolves #2585 - Updates all `core.entity_view_display.node.page.default.yml` instances to remove access to the Hero block. Also attempts to bring these files as much in line as possible by restricting most block access for coronavirus.uiowa.edu and admissions.uiowa.edu, which were based on the collegiate split.
Resolves #2517 - Disables the collegiate split for all sites where its active and removes the split.

# How to test

For the following sites:
- default
- admissions.uiowa.edu (collegiate + site split w/ `core.entity_view_display.node.page.default.yml`)
- cogscilang.grad.uiowa.edu (collegiate)
- coronavirus.uiowa.edu (site split w/ `core.entity_view_display.node.page.default.yml`)
- uiowa.edu (collegiate + site split)

Do the following:
- `blt ds --site=<site>`
- `drush @site.local uli admin/people`
- Check a few pages to ensure they look okay and that you can edit them via Layout Builder
- Masquerade as a webmaster
- Visit a page and click the 'Layout' tab
- Check each available layout to make sure that you can't add the 'Hero' block in any of them
- Repeat as an editor

For collegiate sites also note that there are no issues with the update hook and subsequent config import.

### CODE REVIEW

File: docroot/modules/custom/layout_builder_custom/layout_builder_custom.module [hunk 1]
```diff
@@ -194,11 +194,9 @@ function layout_builder_custom_form_alter(&$form, FormStateInterface $form_state
...
 
         case 'menu_block:main':
-          /** @var \Drupal\Core\Plugin\DefaultPluginManager $filters */
-          $filters = \Drupal::service('plugin.manager.config_filter')->getDefinitions();
-          $collegiate_split = 'config_split:collegiate';
-          // 'collegiate' split either doesn't exist or isn't enabled.
-          if (!isset($filters[$collegiate_split]) || !$filters[$collegiate_split]['status']) {
+          $user = \Drupal::currentUser();
+          // Account for user 1 for local development.
+          if ($user->id() != 1 && !in_array('administrator', $user->getRoles())) {
```
Comment (joewhitsitt) [inline docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: For consistency can we switch this to use docroot/modules/custom/uiowa_core/src/Access/UiowaCoreAccess.php

**Comment by joewhitsitt** [comment]
pulling down for testing

File: docroot/modules/custom/layout_builder_custom/layout_builder_custom.module (hunk 1 above)
Comment (richardbporter) [reply docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: https://github.com/uiowa/uiowa/blob/be7a1b55c9cd983e13a0b8d78627688ec2eb68b2/docroot/modules/custom/uiowa_aggregator/uiowa_aggregator.module#L28-L45

Comment (pyrello) [reply docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: Sounds good to me.

Comment (pyrello) [reply docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: I updated this per the example that @richardbporter supplied.

**Comment by pyrello** [comment]
@joewhitsitt I believe that I have resolved the issue that you raised about the counselor view block not displaying for Admissions.

**Comment by joewhitsitt** [comment]
@pyrello thanks, i was about to push a similar change on Friday but saw the layout column differences in addition to my views change which gave me pause. reason for concern?

**Comment by pyrello** [comment]
As far as I can tell, the rest of the changes are just re-ordering the existing configuration, possibly to put it in alphabetical order? There seems to be a certain amount of fuzziness about how exactly config should be output, so we see these types of changes from time to time with other unrelated exports.

**Comment by joewhitsitt** [comment]
On the default site (v3) and a v2 default site I am getting this during a `ds`:
```
The import failed due to the following reasons: Unexpected error during import with operation update for config_split.config_split.collegiate: The entity does not have an ID.
```

UPDATE: I think I fixed it by looking for the collegiate config regardless of status and deleting it in c15aa43

**Review by joewhitsitt — APPROVED** [review]
Pour one out for collegiate. 🍻 

Tested sync/config import on default, admissions, brand, cogscilang.grad, coronavirus, gis.sites, hr. policy.clas, uiowa. No hiccups after https://github.com/uiowa/uiowa/pull/2586/commits/c15aa435a6461ac9b97ba3fb3977b1f77e8b6813

Spot checked default, v2, v3, old-collegiate sites for Hero.

Confirmed Counselors view was available on Admissions. Not aware of other sites that need certain views for webmasters/editors to place.

**Comment by pyrello** [comment]
I'm going to leave this for now, but I think we may want to switch back to using a deny list for Views, which allows for new views blocks to not have to be strictly added each time.

**Pull request merged**